api/src/pcapi/core/offers/api.py

```diff
@@ -664,6 +667,44 @@ def handle_stocks_edition(edited_stocks: list[typing.Tuple[models.Stock, bool]])
             _notify_beneficiaries_upon_stock_edit(stock, bookings)
 
 
+def update_finance_event_pricing_date(
+    stock: models.Stock | educational_models.CollectiveStock,
+    new_pricing_ordering_date: datetime.datetime,
+    commit: bool = False,
+) -> None:
+    pricing_point_id = stock.offer.venue.current_pricing_point_id
+    finance_events_from_pricing_point = finance_models.FinanceEvent.query.options(
+        sa.orm.joinedload(finance_models.FinanceEvent.pricings)
+    ).filter(
+        finance_models.FinanceEvent.pricingOrderingDate.is_not(None),
+        finance_models.FinanceEvent.pricingPointId == pricing_point_id,
+    )
+
+    booking_ids_subquery = bookings_models.Booking.query.with_entities(bookings_models.Booking.id).filter(
+        bookings_models.Booking.stockId == stock.id
+    )
+    finance_events_from_stock = finance_events_from_pricing_point.filter(
+        finance_models.FinanceEvent.bookingId.in_(booking_ids_subquery)
+    ).all()
+
+    if finance_events_from_stock:
+        old_pricing_ordering_date = finance_events_from_stock[0].pricingOrderingDate
+        pricings_from_pricing_point = []
+        for finance_event in finance_events_from_pricing_point:
+            if finance_event.pricingOrderingDate >= min(old_pricing_ordering_date, new_pricing_ordering_date):
+                finance_event.status = finance_models.FinanceEventStatus.READY
+                pricings_from_pricing_point += finance_event.pricings
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏serait-il possible de mettre un commentaire pour expliquer la règle de gestion sous-jacente ?